sklearn/metrics/tests/test_classification.py

```diff
@@ -153,6 +153,31 @@ def test_classification_report_dictionary_output():
     assert type(expected_report['macro avg']['support']) == int
 
 
+def test_classification_report_output_dict_empty_input():
+    report = classification_report(y_true=[], y_pred=[], output_dict=True)
+    expected_report = {'accuracy': 0.0,
+                       'macro avg': {'f1-score': np.nan,
+                                     'precision': np.nan,
+                                     'recall': np.nan,
+                                     'support': 0},
+                       'weighted avg': {'f1-score': 0.0,
+                                        'precision': 0.0,
+                                        'recall': 0.0,
+                                        'support': 0}}
+    assert isinstance(report, dict)
+    # assert the 2 dicts are equal.
+    assert(report.keys() == expected_report.keys())
```

Comment:
```suggestion
    assert report.keys() == expected_report.keys()
```

Reply:
Does `assert report == expected_report` not work?

Reply:
No that may not always work as the values are float and doing an equal comparison will not be best. Matching keys should be enough, which I do with almost_equal below.

Reply:
Good point. (pytest.approx might work?)